cpp/src/arrow/record_batch_test.cc

```diff
@@ -619,37 +619,24 @@ TEST_F(TestRecordBatch, ConcatenateRecordBatches) {
   ASSERT_BATCHES_EQUAL(*batch, *null_batch);
 }
 
-TEST_F(TestRecordBatch, ToTensorUnsupported) {
+TEST_F(TestRecordBatch, ToTensorUnsupportedType) {
   const int length = 9;
 
```

Comment:
We might want to check bool here too just for clarity, because there is sometimes ambiguity about whether it's numeric or not